test/k8sT/DatapathConfiguration.go

```diff
@@ -319,6 +319,48 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 	})
 
+	Context("IPVLAN", func() {
+		BeforeEach(func() {
+			SkipIfIntegration(helpers.CIIntegrationEKS)
+			SkipIfIntegration(helpers.CIIntegrationFlannel)
+			SkipIfIntegration(helpers.CIIntegrationGKE)
+		})
+
+		It("Check connectivity with IPVLAN", func() {
+			options := map[string]string{
+				"tunnel":              "disabled",
+				"datapathMode":        "ipvlan",
+				"ipvlan.masterDevice": "eth0",
+			}
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
+
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+		})
+
+		It("Check connecitivity with IPVLAN in pure L3 mode", func() {
+			options := map[string]string{
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+				"datapathMode":         "ipvlan",
+				"ipvlan.masterDevice":  "eth0",
+				"installIptablesRules": "false",
+				"l7Proxy.enabled":      "false",
+			}
+			// Needed to bypass bug with masquerading when devices are set. See #12141.
+			if helpers.RunsWithKubeProxy() {
+				options["masquerade"] = "false"
+			}
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
+
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+			if helpers.RunsOnNetNextOr419Kernel() {
+				By("Test BPF masquerade")
+				Expect(testPodHTTPToOutside(kubectl, "http://google.com", false, false)).
+					Should(BeTrue(), "Connectivity test to http://google.com failed")
+			}
```

Comment:
Sorry, I am not really sure if I understand how this is testing BPF masquerading behavior?

Reply:
The BPF masq is currently not supported in the ipvlan mode.

Reply:
OK, then I will just remove that test.